Tests/SentryTests/SentryHubTests.swift

```diff
@@ -204,6 +207,90 @@ class SentryHubTests: XCTestCase {
         }
     }
     
+    func testStartTransactionWithNameOperation() {
+        let span = fixture.getSut().startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
+        let tracer = Dynamic(span)
+        XCTAssertEqual(tracer.name, fixture.transactionName)
+        XCTAssertEqual(span.context.operation, fixture.transactionOperation)
+    }
+    
+    func testStartTransactionWithContext() {
+        let span = fixture.getSut().startTransaction(transactionContext: TransactionContext(name: fixture.transactionName, operation: fixture.transactionOperation))
+        
+        let tracer = Dynamic(span)
+        XCTAssertEqual(tracer.name, fixture.transactionName)
+        XCTAssertEqual(span.context.operation, fixture.transactionOperation)
+    }
+    
+    func testStartTransactionWithContextSamplingContext() {
+        var customSamplingContext: [String: Any]?
+        
+        let options = fixture.options
+        options.tracesSampler = {(context: SamplingContext) -> NSNumber in
+            customSamplingContext = context.customSamplingContext
+            return 0
+        }
+        
+        let span = fixture.getSut().startTransaction(transactionContext: TransactionContext(name: fixture.transactionName, operation: fixture.transactionOperation), customSamplingContext: ["customKey": "customValue"])
+        
+        let tracer = Dynamic(span)
+        XCTAssertEqual(tracer.name, fixture.transactionName)
+        XCTAssertEqual(customSamplingContext?["customKey"] as? String, "customValue")
+        XCTAssertEqual(span.context.operation, fixture.transactionOperation)
+    }
+    
+    func testStartTransactionNotSamplingUsingSampleRate() {
+        let options = fixture.options
+        options.tracesSampleRate = 0.4
+        
+        let hub = fixture.getSut(options)
+        Dynamic(hub).sampler.random = fixture.random
+        
+        let span = hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
+        
+        XCTAssertEqual(span.context.sampled, .no)
+    }
+    
+    func testStartTransactionSamplingUsingSampleRate() {
+        let options = fixture.options
+        options.tracesSampleRate = 0.5
```

Comment:
`m`: Can we also add a test with a wrong value as `-1` and `2`. The same applies to the callback below.